### PR TITLE
Marcin review

### PR DESCRIPTION
**Pull request by dawiddominiak**

### CODE REVIEW

File: What’s up in Vietnam.md [hunk 1]
```diff
@@ -1,62 +1,63 @@
...
 _Laurie Voss_
 
 ### Let’s characterize these two worlds
 
-> Object systems are typically characterized by four basic components: identity, state, behavior and encapsulation.  
+> Object systems are typically characterized by four basic components: identity, state, behavior and encapsulation.
 _Ted Neward_
 
-Summing up the above thought, we can say that every object is described by its identity independently of its state. Every object has a collection of operations that can be invoked to manipulate the state of itself or connected objects, and (most importantly) has encapsulated access to itself and effectively, the ability to inheritance and polymorphism.
+Elaborating on the above though we can say that every object is described by its identity independently of its state. Every object has a collection of operations that can be invoked to manipulate the state of itself or connected objects, and (most importantly) has encapsulated access to itself and effectively, possibility of inheritance and polymorphism.
```
Comment (dawiddominiak) [inline What’s up in Vietnam.md]: I still think that it's ability of an object to inheritance and polymorphism, not a possibility. Can't it stay like it way?

File: What’s up in Vietnam.md [hunk 2]
```diff
@@ -1,62 +1,63 @@
...
 
-> The representation of your object in memory depends what you intend to do with it, and context-sensitive representation is not a feature of OO design.  
+> The representation of your object in memory depends  on what you intend to do with it, and context-sensitive representation is not a feature of OO design.
```
Comment (dawiddominiak) [inline What’s up in Vietnam.md]: I understand you and I can see that mistake, but it's a quote.

File: What’s up in Vietnam.md [hunk 3]
```diff
@@ -1,62 +1,63 @@
...
 
-As we can see, the use of ORM almost always affects the whole application design. It already sounds awful that the whole application depends on one tool. However, the real hell is waiting behind „the repository gate”. 
+As we can see, the use of ORM almost always affects the application design. It already sounds awful that the whole app should depend on one tool. However, the real hell is waiting behind „the repository gate”.
```
Comment (dawiddominiak) [inline What’s up in Vietnam.md]: 1. I wanted to emphasis *whole*. It's the important word here. It's obvious that ORM like every other persistence layer will affect a part of the application design (and application design itself). But it's not obvious that using ORM we need to be careful even in higher abstraction layers such as services or even controllers.

2. IDK if the word `should` is a correct here. It's a fact that `the *whole* application design` depends on ORM and it's a problem.

File: What’s up in Vietnam.md [hunk 4]
```diff
@@ -1,62 +1,63 @@
...
+This is the normalized way to achieve mapping. Every single class maps to a separate table with properties. The `smart_devices` table is in the 1:n relation with `light_management_devices`/`heating_devices`/`security_devices`/`etc. `Tables `light_management_devices`/`heating_devices`/`security_devices` are in 1:n relation with more specific classes. So what’s the disadvantage? Try to create DML to retrieve a record by its ID manually. It’s impractical. We need to also create two artificial primary keys to determine records for more specific tables. Even if it happens behind the scenes, that’s a very inefficient way of retrieving data.
 
-### Table-per-concrete-class approach
+### Table-per-CONCRETE-class approach
```
Comment (dawiddominiak) [inline What’s up in Vietnam.md]: Why capital letters?

File: What’s up in Vietnam.md [hunk 5]
```diff
@@ -65,58 +66,58 @@ UNION Col1, Null as Col2, Col3 FROM heating_devices.
...
 
-In the meantime, our team went to Neo4j conference which was organized by CD PROJECT company. They were creating a significant online card came - Gwent. I realized that they have to have even more significant problems with ranking, so I asked how they deal with it. „We’re using Sphinx” they answered. Then my tech lead asked them: „Why Sphinx? It’s a full-text search engine. How do you deal with a mandatory full-text field in a row?”. „Yeah, but it’s the simplest way to achieve it, and we just leave this mandatory field empty” they answered. I fooled myself.
+In the meantime, our team went to a Neo4j conference, hosted by CD Project game development studio. They were creating a large-scale online card came -- Gwent. I realized that they have to have even more problems with ranking, so I asked how they deal with it. „We’re using Sphinx” they answered. Then my tech lead asked them: „Why Sphinx? It’s a full-text search engine. How do you deal with a mandatory full-text field in a row?”. „Yeah, but it’s the simplest way to achieve it, and we just leave this mandatory field empty” they answered.
```
Comment (dawiddominiak) [inline What’s up in Vietnam.md]: Why sentence "I fooled myself." disappeared?

File: What’s up in Vietnam.md [hunk 6]
```diff
@@ -65,58 +66,58 @@ UNION Col1, Null as Col2, Col3 FROM heating_devices.
...
 2. Over time, composition started winning over inheritance in OOP. Mapping composition to RDBS is easier thanks to its relational origin.
-3. Using the same database by two applications is seen as a bad practice. This and others policy problems mentioned by Ted Neward are a little outdated. But on the other hand, many services are starting cooperatively. Updating such services is a quite hard to achieve. There is a time where one service needs to work with two versions of a database. Forcing ORM framework to work with two different versions of the database is almost impossible in clear and easy to achieve way.
+3. Using the same database by two applications is seen as a bad practice. This and others policy problems mentioned by Ted Neward are a little outdated. But on the other hand, many services are starting to use databases cooperatively. Updating such services is a quite hard. There is a time where one service needs to work with two versions of a database. Forcing ORM framework to work with two different versions of the database is almost impossible in clear and easy to achieve way.
```
Comment (dawiddominiak) [inline What’s up in Vietnam.md]: It lost some sense. My proposition:
"But on the other hand, there are many distributed applications with dozens of services. Updating such services with a database migration is quite hard to achieve."

**Review by dawiddominiak — COMMENTED** [review]
Big thank you for changes <3 . I got a few doubts. Check them and comment or we can shortly discuss.

File: What’s up in Vietnam.md (hunk 1 above)
Comment (ymirpl) [reply What’s up in Vietnam.md]: then it needs to say "ability to inherit and polymorph"

File: What’s up in Vietnam.md (hunk 2 above)
Comment (ymirpl) [reply What’s up in Vietnam.md]: Then the quote had a mistake or was in different context

File: What’s up in Vietnam.md (hunk 4 above)
Comment (ymirpl) [reply What’s up in Vietnam.md]: discard it

File: What’s up in Vietnam.md (hunk 2 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: The whole context:
```
But if your data is not relational,
then you are adding a huge and unnecessary overhead by using
SQL in the first place and then compounding the problem by adding
a further abstraction layer on top of that.

On the the other hand, if your data is relational,
then your object mapping will eventually break down.
SQL is about relational algebra: the output of SQL
is not an object but an answer to a question.
If your object "is" an instance of X and "has" a number of Y,
and each of Y "belongs to" a Z,
what is the correct representation in memory of your object?
Is it merely the properties of X, or should it include all the Ys,
and/or all the Zs? If you get only the properties of X,
when do you run the query to fetch the Ys?
And do you want one or all of them? In reality,
it depends: that's what I mean when I say SQL is the answer to a question.
*The representation of your object in memory depends what you intend to do with it,
and context-sensitive representation is not a feature of OO design.*
Relations are not objects; objects are not relations.
```

File: What’s up in Vietnam.md (hunk 3 above)
Comment (ymirpl) [reply What’s up in Vietnam.md]: We are leaving the "whole" in the second sentence. With "should" it keeps the same meaning, that it's awful that this is happening.

File: What’s up in Vietnam.md (hunk 5 above)
Comment (ymirpl) [reply What’s up in Vietnam.md]: I had no idea what you meant by this sentence and story makes total sense without it.

File: What’s up in Vietnam.md (hunk 6 above)
Comment (ymirpl) [reply What’s up in Vietnam.md]: I like your version more, maybe say "is quite hard task" instead "hard to achieve".

File: What’s up in Vietnam.md (hunk 1 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: I've changed it.

File: What’s up in Vietnam.md (hunk 3 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: I used word `entire`.

File: What’s up in Vietnam.md (hunk 4 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: done

File: What’s up in Vietnam.md (hunk 3 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: I've changed to `I made a fool of myself`.

File: What’s up in Vietnam.md (hunk 5 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: I've changed to `I made a fool of myself`.

File: What’s up in Vietnam.md (hunk 6 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: done

File: What’s up in Vietnam.md (hunk 2 above)
Comment (dawiddominiak) [reply What’s up in Vietnam.md]: Changed to `The representation of your object in memory depends [sic] what you intend to do with it,`

**Pull request merged**